nio/properties/version.py

```diff
@@ -22,9 +22,9 @@ def __init__(self, version=None, title="Version", **kwargs):
                 # hmm, both version and default specified,
                 # version argument overrides
                 kwargs["default"] = version
-            super().__init__(title=title, **kwargs)
+            super().__init__(title=title, advanced=True, **kwargs)
```

Comment:
This should likely be in the actual `version` property declaration in `block/base.py` and `service/base.py` instead of being here.

Reply:
I tried that originally, the issue I was running into was that version is instantiated in the block, ie. https://github.com/nio-blocks/http_requests/blob/1d1518b4212f45f17310f71d153de521faadeb5b/http_requests_base.py#L43 without an `advanced=True` and I think it was overwriting that property in `block/base.py`?

Reply:
Ahh interesting, yeah that makes sense. We'd have to update in every block then, blech. I'm good with setting VersionProperty default to advanced then, this makes sense.

Reply:
what we are saying here is that all `version` properties independently of where they are declared/used will be `advanced`, and cannot be overwritten, don't like it really

Reply:
so it seems like the blocks should not be redefining the property, they should be assigning a value to it in the constructor, is this what you meant @mattdodge here: `We'd have to update in every block then`

Reply:
Really we just want the default for version properties to be `advanced=True`. I suppose if someone created a block with this:
```python
version = VersionProperty('0.1.0', advanced=False)
```
I would want that to override the `True` value. But I don't think we want to update all blocks

Reply:
we need something like this then
```python
        if "advanced" in kwargs:
            advanced = kwargs.pop("advanced")
        else:
            advanced = True
        if "default" in kwargs:
            if version is not None:
                # hmm, both version and default specified,
                # version argument overrides
                kwargs["default"] = version
            super().__init__(title=title, advanced=advanced, **kwargs)
        else:
            super().__init__(title=title, advanced=advanced,
                             default=version, **kwargs)
```

Reply:
I think this would do it too, a bit shorter
```python
advanced = kwargs.get("advanced", True)
```

Reply:
Good discussion guys, it would have to be `kwargs.pop` instead of `.get` otherwise the arg is a duplicate in kwargs, but instead of using `kwargs` I just passed it through as an argument itself.

https://github.com/niolabs/nio/pull/128/commits/feaa5c6828b45673b57cb96d9d6039debcf067d0 allows you to set `version = VersionProperty("0.1.1", advanced=False)` on blocks.